features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMeta.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import com.google.gson.JsonObject
+import java.util.Locale
+import kotlin.jvm.Throws
+
+internal sealed class RumEventMeta {
+
+    abstract val type: String
+
+    open fun toJson(): JsonObject {
+        val model = JsonObject()
+
+        model.addProperty(TYPE_KEY, type)
+
+        return model
+    }
+
+    companion object {
+
+        private const val UNKNOWN_RUM_EVENT_META_TYPE_ERROR = "Unknown RUM event meta type value [%s]"
+
+        const val TYPE_KEY = "type"
+        const val VIEW_TYPE_VALUE = "view"
+        const val VIEW_ID_KEY = "viewId"
+        const val DOCUMENT_VERSION_KEY = "documentVersion"
+
+        @Throws(
+            NullPointerException::class,
+            ClassCastException::class,
+            IllegalStateException::class,
+            NumberFormatException::class
```

Comment:
> [!NOTE]
> It might be relevant to add as a comment what situation those exception could rise from?

Reply:
I changed the way we are doing the parsing, so that it looks now the same as for other RUM events and the only exception thrown is `JsonParseException`. Let me know what you think.